analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Pattern.cs

```diff
@@ -148,15 +161,22 @@ declaration switch
             recursive switch
             {
                 _ when valueConstraint == ObjectConstraint.Null => BoolConstraint.False,
+                _ when RecursivePatternAlwaysMatchesAnyNotNull(recursive) => BoolConstraint.True,
+                _ => null,
+            };
+
+        private static bool RecursivePatternAlwaysMatchesAnyNotNull(IRecursivePatternOperationWrapper recursive) =>
+            recursive switch
```

Comment:
Use `is` expression.

Reply:
Not sure if I like it, but it definitely removes some lines

Reply:
S3101 isn't so happy about this trick, but there's no nice way out of it anyway